main/lsp/requests/hover.cc

```diff
@@ -52,14 +52,24 @@ unique_ptr<ResponseMessage> HoverTask::runRequest(LSPTypecheckerInterface &typec
     }
 
     auto &queryResponses = result.responses;
+    auto clientHoverMarkupKind = config.getClientConfig().clientHoverMarkupKind;
     if (queryResponses.empty()) {
-        // Note: Need to specifically specify the variant type here so the null gets placed into the proper slot.
-        response->result = variant<JSONNullObject, unique_ptr<Hover>>(JSONNullObject());
+        auto fref = config.uri2FileRef(gs, params->textDocument->uri);
+        ENFORCE(fref.exists());
+        auto level = fref.data(gs).strictLevel;
+        if (level < core::StrictLevel::True) {
+            string asString = level == core::StrictLevel::Ignore ? "ignore" : "false";
+            auto text = fmt::format("# No hover information available");
+            auto explanation = fmt::format("The file is `# typed: {}`", asString);
+            response->result = make_unique<Hover>(formatRubyMarkup(clientHoverMarkupKind, text, explanation));
```

Comment:
Is it worth using formatRubyMarkup here?

It seems like somewhat overkill to show

> ```ruby
> # No hover information available
> ```
> ---
> The file is `typed: false`

vs showing

> No hover information available.
> Hint: the file is `typed: false`.